File: languages/java/testData/excludeFromInspection/exclude.java
package testData.excludeFromInspection;

public class Foo {
    public void boo(int boi) {
        var sta<caret>te = Foo.computeSmthg();
        var compo<caret>nent = returnMy();
        var answerForTheUltima<caret>teQuestionOfLife = computeAnswerForTheUltimateQuestionOfLife();
        var s<caret>chool = getSomeSchoolNumber();
        try {
            System.out.println("Oh My!");
        } catch (Exception igno<caret>re) {
        } catch (Exception ign<caret>ored) {
        } catch (Exception x) {
        }
    }

    private State computeSmthg() {
        return new State();
    }

    private int computeAnswerForTheUltimateQuestionOfLife() {
        return 42;
    }

    private int getSomeSchoolNumber() {
        return 239;
    }

    public MyComponent returnMy() {
        return new MyComponent();
    }

    class State();
    class MyComponent();
}

class Foo2 extends Foo {
    @java.lang.Override
    public void boo(int <caret>boi){
        System.out.println("Hello World!");
    }
}

File: languages/kotlin/testData/excludeFromInspection/exclude.kt
package excludeFromInspection

open class Foo {
    open var xy = 0
    open fun boo(boi: Int) {
        val st<caret>ate = computeSmth()
        val compo<caret>nent = returnMy()
        val answerForTheUltimateQu<caret>estionOfLife = computeAnswerForTheUltimateQuestionOfLife()
        try {
            println("Oh My!")
        } catch (ign<caret>ore: Exception) {
        } catch (ig<caret>nored: Exception) {
        } catch (<caret>_: Exception) {
        } catch (e: Exception) {
            println("Exception is not ignored")
        }
    }

    private fun computeSmth(): State {
        return State()
    }

    private fun returnMy(): MyComponent {
        return MyComponent()
    }

    private fun computeAnswerForTheUltimateQuestionOfLife(): Int {
        return 42
    }

    class State
    class MyComponent
}

internal class Foo2 : Foo() {
    override var x<caret>y = 1
    override fun boo(b<caret>oi: Int) = println("Hello World!")
}
